### PR TITLE
Add a multi-stage dockerfile to build console

### PR DESCRIPTION
**Pull request by LinuxSuRen**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
For some users who might don't have the NodeJs environment, it's not easy to build KubeSphere Console. So I provide a multi-stage Dockerfile, people just need to have a Docker environment.

**Which issue(s) this PR fixes**:
None.

**Special notes for reviewers**:
None.

**Additional documentation, usage docs, etc.**:
I already added the related document in the README file.

### CODE REVIEW

File: Dockerfile.multistage [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM node:12.18 as builder
+
+WORKDIR /opt/kubesphere/console
+COPY console.tar.gz .
+RUN tar xzf console.tar.gz
+RUN yarn config set registry https://registry.npm.taobao.org
```
Comment (leoendless) [inline Dockerfile.multistage:6]: No need to use `taobao` registry. `yarn.lock` works fine.

Comment (LinuxSuRen) [reply Dockerfile.multistage:6]: Got you!

File: Dockerfile.multistage [hunk 2]
```diff
@@ -0,0 +1,18 @@
+FROM node:12.18 as builder
```
Comment (leoendless) [inline Dockerfile.multistage:1]: Could the base image be replaced with `node:12-alpine`？

Comment (LinuxSuRen) [reply Dockerfile.multistage:1]: Got some errors below when I change the image from `node:12.18` to `node:12-alpine` :

```
error An unexpected error occurred: "https://registry.yarnpkg.com/core-js/-/core-js-2.6.12.tgz: getaddrinfo EAI_AGAIN registry.yarnpkg.com".
info If you think this is a bug, please open a bug report with the information provided in "/opt/kubesphere/console/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
The command '/bin/sh -c yarn' returned a non-zero code: 1
make: *** [image] Error 1
```

I run command `make image image-push -e REPO=surenpi/ks-console` on the zone `ap2a` of qingcloud.com.

I guess this is a network issue. Right?

Comment (leoendless) [reply Dockerfile.multistage:1]: It seems an issue exists in `node:alpine`. https://github.com/yarnpkg/yarn/issues/6384.

**Pull request merged**